content/src/logic/deployments-queries.ts

```diff
@@ -27,3 +67,169 @@ export async function* streamAllEntityIds(components: Pick<AppComponents, 'datab
+   * then to know where the page should start we will use the lastId.
+   */
+  const pageBorder: string =
+    (order === SortingOrder.ASCENDING ? 'from' : 'to') +
+    (timestampField === SortingField.ENTITY_TIMESTAMP ? 'EntityTimestamp' : 'LocalTimestamp')
+
+  if (filters?.from && timestampField == SortingField.LOCAL_TIMESTAMP) {
+    const fromLocalTimestamp = filters.from
+    if (pageBorder == 'fromLocalTimestamp' && lastId) {
+      whereClause.push(createOrClause('local_timestamp', '>', fromLocalTimestamp, lastId))
+    } else {
+      whereClause.push(SQL`dep1.local_timestamp >= to_timestamp(${fromLocalTimestamp} / 1000.0)`)
+    }
+  }
+  if (filters?.to && timestampField == SortingField.LOCAL_TIMESTAMP) {
+    const toLocalTimestamp = filters.to
+    if (pageBorder == 'toLocalTimestamp' && lastId) {
+      whereClause.push(createOrClause('local_timestamp', '<', toLocalTimestamp, lastId))
+    } else {
+      whereClause.push(SQL`dep1.local_timestamp <= to_timestamp(${toLocalTimestamp} / 1000.0)`)
+    }
+  }
+
+  if (filters?.from && timestampField == SortingField.ENTITY_TIMESTAMP) {
+    const fromEntityTimestamp = filters.from
+    if (pageBorder == 'fromEntityTimestamp' && lastId) {
+      whereClause.push(createOrClause('entity_timestamp', '>', fromEntityTimestamp, lastId))
+    } else {
+      whereClause.push(SQL`dep1.entity_timestamp >= to_timestamp(${fromEntityTimestamp} / 1000.0)`)
+    }
+  }
+  if (filters?.to && timestampField == SortingField.ENTITY_TIMESTAMP) {
+    const toEntityTimestamp = filters.to
+    if (pageBorder == 'toEntityTimestamp' && lastId) {
+      whereClause.push(createOrClause('entity_timestamp', '<', toEntityTimestamp, lastId))
+    } else {
+      whereClause.push(SQL`dep1.entity_timestamp <= to_timestamp(${toEntityTimestamp} / 1000.0)`)
+    }
+  }
+
+  if (filters?.deployedBy && filters.deployedBy.length > 0) {
+    const deployedBy = filters.deployedBy.map((deployedBy) => deployedBy.toLocaleLowerCase())
+    whereClause.push(SQL`LOWER(dep1.deployer_address) = ANY (${deployedBy})`)
+  }
+
+  if (filters?.entityTypes && filters.entityTypes.length > 0) {
+    const entityTypes = filters.entityTypes
+    whereClause.push(SQL`dep1.entity_type = ANY (${entityTypes})`)
+  }
+
+  if (filters?.entityIds && filters.entityIds.length > 0) {
+    const entityIds = filters.entityIds
+    whereClause.push(SQL`dep1.entity_id = ANY (${entityIds})`)
+  }
+
+  if (filters?.onlyCurrentlyPointed) {
+    whereClause.push(SQL`dep1.deleter_deployment IS NULL`)
+  }
+
+  if (filters?.pointers && filters.pointers.length > 0) {
+    const pointers = filters.pointers.map((p) => p.toLowerCase())
+    whereClause.push(SQL`dep1.entity_pointers && ARRAY[${pointers}]`)
+  }
+
+  const where =
+    whereClause.length > 0
+      ? SQL` WHERE `.append(
+          whereClause.slice(1).reduce((previous, current) => {
+            return previous.append(' AND ').append(current)
+          }, whereClause[0])
+        )
+      : ''
```

Comment:
Can you take the chance and make this be more happy to read?

Reply:
This whole query needs that, but maybe in another PR?